src/coreclr/gc/gc.cpp

```diff
@@ -25152,37 +25165,86 @@ float median_of_3 (float a, float b, float c)
     return b;
 }
 
-size_t gc_heap::get_num_completed_gcs ()
+float log_with_base (float x, float base)
 {
```

Comment:
it's actually meant to have x > base and should be enforced. but I can still add an assert.

Reply:
`log_b(x)` is fine for `x <= base` (e.g., log_2(2) = 1, log_4(2) = 1/2)

I think you're saying (by "should be enforced") that current call site(s) expect `x > base`.  `log_with_base` is a very reasonable helper function that could get used elsewhere without such a restriction.  Or maybe you want to rename it to show that it is intended as a helper for a specific context rather than a general log helper?